plugins/vault-backend/src/service/vaultApi.ts

```diff
@@ -115,48 +124,109 @@ export class VaultClient implements VaultApi {
     return `${this.vaultConfig.baseUrl}/ui/vault/secrets/${this.vaultConfig.secretEngine}`;
   }
 
+  async processMetadataResponse(
```

Comment:
You declared `VaultSecretList` as `@internal`, so now you probably have to declare this private; `VaultClient` is public so you can't have internals being visible.

Reply:
Thanks for your feedback, will have a look at it.
BTW I changed this PR to a draft anyway, since I'd like to change the behavior to optionally enable subkeys requests in addition to the metadata requests (instead of either/or as I have it now).
This way it would be possible to discover multiple secret entries as it was already possible using the metadata implementation and optionally show their subkeys, if somebody wants to.